### PR TITLE
=str - Makes GraphStageWithMaterializedValue covariant in the Shape p…

### PR DESCRIPTION
**Pull request by viktorklang**
…arameter and removes casts

### CODE REVIEW

**Comment by drewhk** [comment]
ah, makes sense

**Comment by ktoso** [comment]
LGTM

File: akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala [hunk 1]
```diff
@@ -12,8 +12,7 @@ import scala.collection.{ immutable, mutable }
 import scala.concurrent.duration.FiniteDuration
 import scala.collection.mutable.ArrayBuffer
 
-abstract class GraphStageWithMaterializedValue[S <: Shape, M] extends Graph[S, M] {
-  def shape: S
+abstract class GraphStageWithMaterializedValue[+S <: Shape, M] extends Graph[S, M] {
```
Comment (rkuhn) [inline akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala]: While we’re at it: M should also be covariant.

**Comment by rkuhn** [comment]
LGTM after tiny fix.

File: akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala [hunk 2]
```diff
@@ -12,8 +12,7 @@ import scala.collection.{ immutable, mutable }
 import scala.concurrent.duration.FiniteDuration
 import scala.collection.mutable.ArrayBuffer
 
-abstract class GraphStageWithMaterializedValue[S <: Shape, M] extends Graph[S, M] {
-  def shape: S
+abstract class GraphStageWithMaterializedValue[+S <: Shape, +M] extends Graph[S, M] {
```
Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala]: @rkuhn Done

**Comment by drewhk** [comment]
LGTM, Please rebase

**Comment by viktorklang** [comment]
@drewhk Done

**Pull request merged**